packages/snaps-controllers/src/snaps/SnapController.test.tsx

```diff
@@ -5330,6 +5330,242 @@ describe('SnapController', () => {
+        'PermissionController:getPermissions',
+        () => ({}),
+      );
+
+      rootMessenger.registerActionHandler(
+        'SelectedNetworkController:getNetworkClientIdForDomain',
+        () => 'mainnet',
+      );
+
+      rootMessenger.registerActionHandler(
+        'NetworkController:getNetworkClientById',
+        () => ({
+          configuration: {
+            chainId: '0x1',
+          },
+        }),
+      );
+
+      const { manifest } = await getMockSnapFilesWithUpdatedChecksum({
+        manifest: getSnapManifest({
+          initialPermissions: {
+            'endowment:page-home': {},
+            'endowment:ethereum-provider': {},
+          },
+        }),
+      });
+
+      const snapController = getSnapController(
+        getSnapControllerOptions({
+          messenger,
+          detectSnapLocation: loopbackDetect({ manifest }),
+        }),
+      );
+
+      await snapController.installSnaps(MOCK_ORIGIN, {
+        [MOCK_SNAP_ID]: {},
+      });
+
+      const approvedPermissions = {
+        'endowment:page-home': {
+          caveats: null,
+        },
+        'endowment:ethereum-provider': {},
+        'endowment:caip25': {
+          caveats: [
+            {
+              type: 'authorizedScopes',
+              value: {
+                requiredScopes: {},
+                optionalScopes: {
+                  'eip155:1': {
+                    accounts: [],
+                  },
+                },
+                sessionProperties: {},
+                isMultichainOrigin: false,
+              },
+            },
+          ],
+        },
+      };
+
+      expect(messenger.call).toHaveBeenCalledWith(
+        'PermissionController:grantPermissions',
+        {
+          approvedPermissions,
+          subject: { origin: MOCK_SNAP_ID },
+          requestData: expect.any(Object),
+        },
+      );
+
+      snapController.destroy();
+    });
+
+    it('overrides the `endowment:caip25` permission if the Snap specifies it in its manifest', async () => {
+      const rootMessenger = getControllerMessenger();
+      const messenger = getSnapControllerMessenger(rootMessenger);
+
+      rootMessenger.registerActionHandler(
+        'PermissionController:getPermissions',
+        () => ({}),
+      );
+
+      rootMessenger.registerActionHandler(
+        'SelectedNetworkController:getNetworkClientIdForDomain',
+        () => 'mainnet',
+      );
+
+      rootMessenger.registerActionHandler(
+        'NetworkController:getNetworkClientById',
+        () => ({
+          configuration: {
+            chainId: '0x1',
+          },
+        }),
+      );
+
+      const { manifest } = await getMockSnapFilesWithUpdatedChecksum({
+        manifest: getSnapManifest({
+          initialPermissions: {
+            'endowment:page-home': {},
+            'endowment:ethereum-provider': {},
+            // @ts-expect-error: Not allowed in `SnapManifest` type.
+            'endowment:caip25': {
```

Comment:
This is disallowed via `excludedPermissions` in extension, so it cannot be requested at install time

Reply:
Should I just remove this test then?

Reply:
IMO yes